protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/MailCmdHandler.java

```diff
@@ -233,34 +232,40 @@ private Response doMAILFilter(SMTPSession session, String argument) {
                 }
 
                 try {
-                    senderAddress = new MailAddress(sender);
+                    senderAddress = MaybeSender.of(new MailAddress(sender));
                 } catch (Exception pe) {
                     LOGGER.info("Error parsing sender address: {}", sender, pe);
                     return SYNTAX_ERROR_ADDRESS;
                 }
             }
-            if ((senderAddress == null) || 
-                    ((senderAddress.getLocalPart().length() == 0) && (senderAddress.getDomain().name().length() == 0))) {
-                senderAddress = MailAddress.nullSender();
+            if (isNullSender(senderAddress)) {
+                senderAddress = MaybeSender.nullSender();
             }
             // Store the senderAddress in session map
             session.setAttachment(SMTPSession.SENDER, senderAddress, State.Transaction);
         }
         return null;
     }
-    
+
+    private boolean isNullSender(MaybeSender senderAddress) {
+        if (senderAddress.isNullSender()) {
+            return true;
+        }
+        boolean hasEmptyLocalPart = senderAddress.get().getLocalPart().length() == 0;
```

Comment:
this doesn't make sense.
What is the difference between MaybeAdresse.isNullSender and isNullSender ?

Reply:
The new version checks the `<@>` case before creating the MaybeSender stuff